django/db/backends/postgresql/operations.py

```diff
@@ -270,3 +270,28 @@ def fulltext_search_sql(self, field_name):
             "Add 'django.contrib.postgres' to settings.INSTALLED_APPS to use "
             "the search operator."
         )
+
+    def window_frame_range_start_end(self, start=None, end=None):
+        """
+        PostgreSQL disallows RANGE BETWEEN n PRECEDING AND n FOLLOWING,
+        """
+        if not isinstance(start, (int, type(None))) and isinstance(end, (int, type(None))):
+            raise TypeError('Arguments must be of type int or None')
```

Comment:
You do more type checking below, you could put this error into those clauses instead of doing it twice